app/routes/messages.py

```
@@ -13,16 +20,25 @@
     "/api/v1/messages", dependencies=[Depends(PermissionsValidator())]
 )
 def send_message(body: SendMessageRequest) -> SendMessageResponse:
+    try:
+        model = LLMModel(body.preferred_model)
+    except ValueError as e:
+        raise InvalidModelException(str(e))
+
     guidance = GuidanceWrapper(
-        model=body.preferred_model,
+        model=model,
         handlebars=body.template.content,
         parameters=body.parameters,
     )
 
     try:
         content = guidance.query()
-    except (KeyError, ValueError) as e:
-        raise BadDataException(str(e))
+    except KeyError as e:
+        raise MissingParameterException(str(e))
+    except (SyntaxError, IncompleteParseError) as e:
+        raise InvalidTemplateException(str(e))
+    except BaseException as e:
```

do u intend to use BaseException? It will catch some other "low level" error like KeyboardInterupt or SystemExit. In this extreme case case I think we dont need to catch and just let it fail so dev can have proper action. We can use Exception instead

---

I actually want to catch everything that leads to no message being returned and report it back to Artemis properly. 
Not sure what is the better option for this here.

---

I think `Exception` will cover most of the cases already. If we catch for example SystemExit without raising it back, the system will hang forever